pkg/netpol/eval/resources.go

```diff
@@ -36,6 +38,10 @@ type (
 		netpolsMap                      map[string]map[string]*k8s.NetworkPolicy // map from namespace to map from netpol name to its object
 		podOwnersToRepresentativePodMap map[string]map[string]*k8s.Pod           // map from namespace to map from pods' ownerReference name
 		// to its representative pod object
+		adminNetpolsMap    map[string]bool           // set of input admin-network-policies names to ensure uniqueness by name
+		sortedAdminNetpols []*k8s.AdminNetworkPolicy // sorted by priority list of admin-network-policies;
+		// sorting ANPs occurs after all input k8s objects are inserted to the policy-engine
+		baselineAdminNetpol    *k8s.BaselineAdminNetworkPolicy // pointer to BaselineAdminNetworkPolicy which is a cluster singleton object
```

Comment:
where is it documented in the spec that `BaselineAdminNetworkPolicy` is a singleton?

the documentation of `BaselineAdminNetworkPolicyIngressRule` states that `A total of 100 Ingress rules will be allowed in each BANP instance.` , which may hint there could be more than one instance?

Reply:
1. [go resources](https://pkg.go.dev/sigs.k8s.io/network-policy-api@v0.1.5/apis/v1alpha1#BaselineAdminNetworkPolicy)
see in the description:
> message="Only one baseline admin network policy with metadata.name=\"default\" can be created in the cluster" BaselineAdminNetworkPolicy is a cluster level resource that is part of the AdminNetworkPolicy API. 

2. from [openshift-resource](https://www.redhat.com/en/blog/using-adminnetworkpolicy-api-to-secure-openshift-cluster-networking)

> The BaselineAdminNetworkPolicy resource is a cluster singleton object and acts mostly as the default catch-all guardrail policy in case a passed traffic does not match any NetworkPolicies in the cluster. You must use default as the name when creating a BaselineAdminNetworkPolicy object.